torch/lib/THC/THCTensorIndex.cu

```diff
@@ -332,5 +332,63 @@ __global__ void indexSelectLargeIndex(TensorInfo<T, IndexType> dst,
   }
 }
 
+template <typename IndexType, unsigned int Dims>
+struct LinearIndexCalcData {
+  // sizes for Tensor dims (either from the Tensor, or the size of the adv indexer at that dim)
+  IndexType sizes[Dims];
+  IndexType strides[Dims];       // strides for Tensor
+  bool adv[Dims];                // which Tensors are advanced indexers
+  long *advIndexTensors[Dims];   // Adv Indexing Tensors
```

Comment:
these comments aren't really helpful. About `adv` - don't Tensors *always* trigger advanced indexing (also `is_adv` might be a better name)? Also, what's `advIndexTensors`? It's impossible to tell both from the name and the comment. Is it their data or their sizes?

Also, isn't it enough to use NULL pointers in this array instead of storing `adv` separately?